File: check_provinces.js
const fs = require('fs');
const content = fs.readFileSync('js/data.js', 'utf8');
const jsonStr = content.replace(/^\s*var\s+UDON_COMPANIES\s*=\s*/, '').replace(/;\s*$/, '');
const companies = JSON.parse(jsonStr);

console.log('Total companies in data.js:', companies.length);

const provMap = {};
const nonUdon = [];

companies.forEach(c => {
  const p = c.province || 'UNKNOWN';
  provMap[p] = (provMap[p] || 0) + 1;
  if (p !== 'อุดรธานี') {
    nonUdon.push(c);
  }
});

console.log('=== Province Breakdown ===');
console.log(provMap);

console.log('=== Companies NOT in อุดรธานี ===');
nonUdon.forEach(c => {
  console.log(`ID: ${c.id} | Name: ${c.name} | Province: ${c.province} | District: ${c.district}`);
});

File: check_provinces.py
import json
import re

with open("js/data.js", "r", encoding="utf-8") as f:
    text = f.read()

# remove `var UDON_COMPANIES = ` and trailing `;`
text = re.sub(r'^\s*var\s+UDON_COMPANIES\s*=\s*', '', text)
text = re.sub(r';\s*$', '', text)

data = json.loads(text)

print(f"Total companies in data.js: {len(data)}")

prov_map = {}
non_udon = []

for c in data:
    prov = c.get("province", "UNKNOWN")
    prov_map[prov] = prov_map.get(prov, 0) + 1
    if prov != "อุดรธานี":
        non_udon.append(c)

print("=== Province Breakdown ===")
for p, count in prov_map.items():
    print(f"{p}: {count} companies")

print("\n=== Companies NOT in อุดรธานี ===")
for c in non_udon:
    print(f"ID: {c.get('id')} | Name: {c.get('name')} | Province: {c.get('province')} | District: {c.get('district')}")
